aws_lambda_powertools/utilities/batch/sqs.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+
+"""
+Batch SQS utilities
+"""
+
+from typing import List
+
+import boto3
+
+from aws_lambda_powertools.middleware_factory import lambda_handler_decorator
+
+from .base import BasePartialProcessor
+
+
+class BasePartialSQSProcessor(BasePartialProcessor):
+    def __init__(self):
+        self._client = boto3.client("sqs")
+        self.success_messages: List = []
+        self.fail_messages: List = []
+
+        super().__init__()
+
+    def get_queue_url(self):
+        *_, account_id, queue_name = self.records[0]["eventSourceARN"].split(":")
+        return f"{self._client._endpoint.host}/{account_id}/{queue_name}"
+
+    def get_entries_to_clean(self):
+        return [{"Id": msg["messageId"], "ReceiptHandle": msg["receiptHandle"]} for msg in self.success_messages]
+
+    def _process_record(self, record):
+        try:
+            result = self.handler(record)
+            return self.success_handler(record, result)
+        except Exception as exc:
+            return self.failure_handler(record, exc)
+
+    def _prepare(self):
+        """
+        """
+        self.success_messages.clear()
+        self.fail_messages.clear()
+
+    def _clean(self):
+        """
+        """
+        if not self.fail_messages:
+            return
+
+        queue_url = self.get_queue_url()
+        entries_to_remove = self.get_entries_to_clean()
+
+        return self._client.delete_message_batch(QueueUrl=queue_url, Entries=entries_to_remove)
+
+
+class DefaultPartialSQSProcessor(BasePartialSQSProcessor):
+    pass
+
+
+@lambda_handler_decorator
+def partial_sqs_processor(handler, event, context, record_handler, processor=None):
+    records = event["Records"]
+    processor = processor or DefaultPartialSQSProcessor()
+
+    with processor(records, record_handler) as ctx:
+        ctx.process()
```

Comment:
That's a very good point, let me try to explain. I'm not calling `handler` directly because this decorator must receive a `record_handler`, a Callable which knows how to handle a single record, such that your implementation would look something like that:
```python
def pass_through_handler(record):
   return record

@lambda_handler_decorator(pass_through_handler)
def handler(event, context):
  pass
```
The computation will be done by this `record_handler`, not the original handler. 

BTW you may have noticed a comment in `BaseProcessor`. I've implemented `__call__` to allow multiple runs of the same processor but with different data or handler. One missing param is`context`, do you see any use case where user may change it during execution ?

Reply:
The `@lambda_handler_decorator` middleware allows you to create a decorator which in turn can be used to decorate a lambda handler function. As such, you're expected to call the lambda handler - I think I understand your intent, but with the implementation as it stands right now, the lambda handler won't actually be called at all. To use your example:

```
def pass_through_handler(record):
   return record  # this will be executed for each SQS record

@lambda_handler_decorator(pass_through_handler)
def handler(event, context):
  other_logic()  # Not executed
  return {"statusCode": 200, "result": "whatever"}  # Not executed
```

With regards to the lambda context, I can't think of a scenario where the record processor would need knowledge of it so I think its fine to leave it out.

Reply:
Nice catch !! I'll call the handler

Reply:
So looking at this we now have a decorator that will be invoked and process all received records with the given `record_handler` method the return value is captured by the `PartialSQSProcessor` class but not passed to the original handler.

In other words, it does not make sense to store the results in Lists if we are not providing the results to a consumer. So either we inject the results of the processed messages in the event before calling the handler or we can simplify the logic for failures to a boolean flag.

I think from a flexibility point of view you want to enrich the event? Thoughts anyone?

Reply:
Yeap ! It works like a charm if we run as a context. I was expecting someone to raise this discussion, thx ! 

My initial tought was to enrich the event, but I'm not a huge fan of this approach since we're going to change the interface of the integration. I'm more inclined to follow a "kwargs" approach.

Opnions would be great !

Reply:
So how would the `kwargs` approach look like, something like this then?

```pyhton
def pass_through_handler(record):
   return {"Foo": "Bar"}

@lambda_handler_decorator(pass_through_handler)
def handler(event, context, **kwargs):
  assert kwargs["record_results"][0] == {"Foo": "Bar"}
```

Reply:
@heitorlessa , @cakepietoast. Can you please take a look ?